File: Laborator 5/Laborator/ExecutorService/app/src/main/java/com/example/simpleasynctask/MainActivity.java
package com.example.simpleasynctask;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class MainActivity extends AppCompatActivity {

    private TextView textResult;
    private Button buttonStart;
    private ExecutorService executorService;
    private Handler handler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textResult = findViewById(R.id.text_result);
        buttonStart = findViewById(R.id.button_start);

        executorService = Executors.newSingleThreadExecutor();
        handler = new Handler(Looper.getMainLooper());

        buttonStart.setOnClickListener(v -> {
            textResult.setText("Sarcina a început...");
            executorService.execute(() -> {
                // Simulăm o sarcină de lungă durată
                for (int i = 1; i <= 5; i++) {
                    try {
                        Thread.sleep(1000); // Așteaptă 1 secundă
                        int progress = i * 20;
                        handler.post(() -> textResult.setText("Progres: " + progress + "%"));
                    } catch (InterruptedException e) {
                        e.printStackTrace();
                    }
                }
                handler.post(() -> textResult.setText("Sarcina s-a terminat!"));
            });
        });
    }
}

File: Laborator 6/Laborator/MyKtlApplication/app/src/main/java/ro/cunbm/myktlapplication/MainActivity.kt
package ro.cunbm.myktlapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {
    val TAG ="MainActivity"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // not here (outside of coroutine): doNetworkCall()
        GlobalScope.launch{
            delay(1000L)
            Log.d(TAG, "Coroutine says Hello from thread ${Thread.currentThread().name}")
            val networkCallAnswer = doNetworkCall()
            val networkCallAnswer2 = doNetworkCall2()
            Log.d(TAG, networkCallAnswer)
            Log.d(TAG, networkCallAnswer2)
        }
        Log.d(TAG, "Hello from thread ${Thread.currentThread().name}")
        Thread.sleep(2000L);
    }

    suspend fun doNetworkCall():String{
        delay(3000L)
        return "\n*** This is the message 1\n"
    }

    suspend fun doNetworkCall2():String{
        delay(3000L)
        return "\n*** This is the message 2\n"
    }
}